core/domain/exp_jobs_one_off.py

```diff
@@ -301,6 +301,41 @@ def reduce(key, values):
         yield (key, values)
 
 
+class ExplorationMathTagValidationOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that checks the html content of an exploration and validates all the
+    Math tags in the HTML.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        exploration = exp_fetchers.get_exploration_from_model(item)
+        for state_name, state in exploration.states.items():
+            html_string = ''.join(state.get_all_html_content_strings())
+            error_list = (
+                html_validation_service.validate_math_tags_in_html(html_string))
+            if len(error_list) > 0:
+                yield ({
+                    'exp_id': item.id,
```

Comment:
You don't need the exp rights, I think? Or if you need specific fields from that, just yield the specific fields.

Also you should not be fetching models in a for loop. Do it once at the outset instead. Always think about performance.

Reply:
> You don't need the exp rights, I think? Or if you need specific fields from that, just yield the specific fields.

@DubeySandeep suggested that I might not be allowed to use the data from private explorations, that is the reason I added this. Should I still keep it or is it not needed?.

> Also you should not be fetching models in a for loop. Do it once at the outset instead

I didn't completely get which lines are you referring to here, can you mention which lines?.

Reply:
`rights_manager.get_exploration_rights` makes DataStore call and you have this code in for loop, fetch the status before the for loop! is that helpful?

Reply:
done